File: js/src/maplestory/api/dto/characterAndroidCashItemEquipmentColoringPrismDto.ts
import { CharacterAndroidCashItemEquipmentColoringPrismDtoBody } from '../response/characterAndroidEquipmentDtoBody';

/**
 * 캐릭터 안드로이드 캐시 아이템 컬러링프리즘 정보
 */
class CharacterAndroidCashItemEquipmentColoringPrismDto {
  /**
   * 컬러링프리즘 색상 범위
   */
  colorRange: string;

  /**
   * 컬러링프리즘 색조
   */
  hue: number;

  /**
   * 컬러링프리즘 채도
   */
  saturation: number;

  /**
   * 컬러링프리즘 명도
   */
  value: number;

  constructor(obj: CharacterAndroidCashItemEquipmentColoringPrismDtoBody) {
    const { color_range, hue, saturation, value } = obj;

    this.colorRange = color_range;
    this.hue = hue;
    this.saturation = saturation;
    this.value = value;
  }
}

export { CharacterAndroidCashItemEquipmentColoringPrismDto };

File: java/src/main/java/dev/spiralmoon/maplestory/api/msea/dto/character/CharacterAndroidCashItemEquipmentColoringPrismDTO.java
package dev.spiralmoon.maplestory.api.msea.dto.character;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

/**
 * Android cash item coloring prism information
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@ToString
public class CharacterAndroidCashItemEquipmentColoringPrismDTO implements dev.spiralmoon.maplestory.api.common.dto.character.CharacterAndroidCashItemEquipmentColoringPrismDTO {

    /**
     * Coloring prism color range
     */
    @JsonProperty("color_range")
    private String colorRange;

    /**
     * Coloring prism hue
     */
    @JsonProperty("hue")
    private Long hue;

    /**
     * Coloring prism saturation
     */
    @JsonProperty("saturation")
    private Long saturation;

    /**
     * Coloring prism brightness
     */
    @JsonProperty("value")
    private Long value;
}
